CONTRIBUTING.md

```diff
@@ -11,4 +11,116 @@ visualization.
+your idea has not already been implemented. 
+
+Once you have an example you would like to add there are a few guide lines to follow.
+Every example should:
+- be saved as a stand alone script in the `altair/examples/` directory.
+- have a descriptive docstring, which will eventually be extracted for the
+  documentation website.
+- contain a category tag.
+- define a chart variable with the main chart object (This will be used both in
+  the unit tests to confirm that the example executes properly, and also
+  eventually used to display the visualization on the documentation website).
+- not make any external calls to download data within the script (i.e. don't
+  use urllib). You can define your data directly within the example file,
+  generate your data using pandas and numpy, or you can use data
+  available in the `vega_datasets` package.
+
+The easiest way to get started would be to adapt examples from the [Vega-Lite
+example gallery](https://vega.github.io/vega-lite/examples/) which are missing
+in the altair gallery. Or you can feel free to be creative and build your own
+visualizations.
+
+
+#### Initial Setup
+Install the latest version of Altair locally using 
+```
+pip install git+https://github.com/altair-viz/altair/
+```
+Next step is to fork the repository on GitHub and clone the fork to you local
+machine. For more details on forking see the [GitHub
+Documentation](https://help.github.com/en/articles/fork-a-repo).
+```
+git clone https://github.com/YOUR-USERNAME/altair.git
+```
+You can have a single clone of the repository that points to both your fork and
+the main package repository. These pointers to GitHub are called "remotes" .
+On your local clone you should run:
+```
+git remote add upstream https://github.com/altair-viz/altair.git
+git checkout master
+git pull upstream master
+```
+And then you'll have all the updates in the master branch of your local fork.
+Note that git will complain if you've committed changes to your local master
+branch that are not on upstream (this is one reason it's good practice to **never**
+work directly on your master branch). 
+
+#### Steps to Contribute
+
+Often it is convenient to draft an example outside of the main repository, such
+as [Google Colab](https://colab.research.google.com/), to avoid difficulties
+when working with git. Once you have an example you would like to add the procedure is
+to first update your master branch. 
+```
+git checkout master
+git pull upstream master
+```
+Then create a branch for your example submission
+```
+git checkout -b <branch-name>
+```
+Now you are on a new branch where you can add your example file. All examples
+files are kept in the `altair/examples/` directory. Make sure the style and
+format match the other examples.  An official style guide for examples has not
+been created yet, but a few suggestions are 
+- Underline the title in the docstring with either `---` or `===`. Try to match
+  the number of characters. 
```

Comment:
More specifically, something like "The file docstring  will be rendered into HTML via [reStructuredText](http://docutils.sourceforge.net/rst.html), so use that format for any hyperlinks or text styling. In particular, be sure you include a title in the docstring underlined with `---`, and be sure that the size of the underline exactly matches the size of the title text."